### ISSUE TITLE
Ventilation rate procedure (VRP) sizing problems

### ISSUE THREAD

**Issue by mjwitte** [NREL/EnergyPlus#6055]
Issue overview
--------------
I tried the autosize feature on the reheat box min fraction and it does work but I’m still not getting enough OA out of the system. Here is the problem: when the system OA is upsized based on 62.1, i.e., the system OA is greater than the zone sum, the minimum damper positions required to support the higher OA are the ones that should be used. Further, the max supply for some zones may need to be upsized to support the higher system OA.

In the Primary School example, the system OA is sized to 3.82 m3/s by the VRP sizing, which is much higher than the zone sum OA, but the autosized minimum damper positions ensure only the breathing zone OA will be provided (assuming those rates were input in the zone sizing object). This means under certain conditions, zones could receive less OA than the breathing minimum.

Regarding your question about an example of where the zone maximum is not upsized, see the zone MULT_CLASS_2_POD_1_ZN_1_FLR_1.  It’s max supply should be upsized to equal the breathing zone OA divided by 0.6. This means it’s maximum supply will be greater than what it was sized during zone sizing.  Minimum supply m3/s = 0.9086/0.6 = 1.51 m3/s, which is greater than the zone max supply of 1.123 m3/s calculated earlier.

We need to configure the minimum damper positions to ensure that the mixed flow rate is at least equal to the autosized VRP 62.1 system OA flow rate.

Here is my suggestion to allow this operation in V8.7:

1. Check if the current minimum damper positions can support the new system OA calculated using autosized VRP and reset the positions as necessary. Carry forward this information to the reheat box minimum. 
*Currently, autosizing of the min flow rate for a VAV damper does not pay any attention to OA requirements.  It is based on the Sizing:Zone input fields ”Cooling Minimum Air Flow per Zone Floor Area”, ”Cooling Minimum Air Flow”, and ”Cooling Minimum Air Flow Fraction”. If there is no sizing calculation a default of .000762 cubic meters per second per square meter of zone floor area (0.15 cfm/ft2) is used.*

2. Check if there are zones that need more supply than the maximum and reset the maximum as necessary. 

3.  When DesignSpecification:OutdoorAir is entered in the VAV terminal unit, the control does not account for the ventilation efficiency. Need new input(s) in the terminal unit?

4.  Also look at impact of Sizing:System     1.0,                     !- Zone Maximum Outdoor Air Fraction {dimensionless}

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version)
 - Version of EnergyPlus (if using an intermediate build, include SHA)
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

**Comment by mjwitte** [comment]
Relevant inputs:
```
  DesignSpecification:ZoneAirDistribution,
    Class_POD_DSZAD,         !- Name
    1,                       !- Zone Air Distribution Effectiveness in Cooling Mode {dimensionless}
    1,                       !- Zone Air Distribution Effectiveness in Heating Mode {dimensionless}
    ,                        !- Zone Air Distribution Effectiveness Schedule Name
    ,                        !- Zone Secondary Recirculation Fraction {dimensionless}
    0.6;                    !- Minimum Zone Ventilation Efficiency {dimensionless}

  Sizing:System,
    VAV_POD_1,               !- AirLoop Name
    Coincident,              !- Type of Zone Sum to Use
    VentilationRateProcedure,!- System Outdoor Air Method
    1.0,                     !- Zone Maximum Outdoor Air Fraction {dimensionless}

```

1.17.3.1.6 Field: Minimum Zone Ventilation Efficiency
This optional input sets a minimum on the ventilation efficiency for the zone. It is only used with the Ventilation Rate Procedure (VRP), single-path method. VRP should be chosen in Sizing System, System Outdoor Air Method = VentilationRateProcedure. Single-path method is indicated by leaving the previous input (Zone Secondary Recirculation Fraction) blank or setting it to 0.0. If the calculated value of ventilation efficiency for a zone is less than this value, it is raised to this minimum by raising the zone minimum air flow rate. This new value for zone minimum air flow rate then overrides other defaults and inputs in Sizing:Zone.

From the Engineering Ref:
![image](https://cloud.githubusercontent.com/assets/4991075/24971792/ca76bc26-1f7e-11e7-9458-cf9039d1bd73.png)


From a related source:
The auto-calculated EvZn for each zone should be based on Equations G-1 and G-2 from Std 62.1 depending on whether the air terminal is single supply (e.g., VAV, VRH) or recirculated (e.g, SFPVAV, PFPVAV)
```
    IF (SingleSupplyFlag == .TRUE.) THEN
        !Single Supply, i.e., Ventilation not re-mixed before supply air
        !  is dumped into space
        EvZn = MIN(1.0, 1.0 + Xs - Zd)                                                      !Eq G-1
    ELSE
        !General case: Ventilation possibly re-mixed before supply air
        !  is dumped into space, i.e, 2FDDVV, PFPVAV, SFPVAV, TAB
        !  This eq is same as  Ev = (Fa + Xs*Fb - Zpz*Ep*Fc) / Fa
        EvZn = MIN(1.0, (Fa + Xs*Fb - Zd * Fc) / Fa)                                 !Eq G-2
    ENDIF
```

These equations agree with the EnergyPlus source, except EnergyPlus does not enforce a max value of 1.0 for Evz (not sure if this matters).

**Comment by EnergyArchmage** [comment]
I think that coincident vs noncoincident is also an important distinction for these VRP sizing problems.

**Comment by mjwitte** [comment]
Comparison of results for defect file with Minimum Zone Ventilation Efficiency = 0.6 vs 0.7 for VAV_POD_1 (note that there is also a corridor zone on POD1 - not shown here)

![image](https://cloud.githubusercontent.com/assets/4991075/24972189/1fb43fdc-1f80-11e7-8945-e5a60eb83616.png)
![image](https://cloud.githubusercontent.com/assets/4991075/24972206/2cb0172e-1f80-11e7-8bb1-eb6168d9ad22.png)

![image](https://cloud.githubusercontent.com/assets/4991075/24972308/9dcfd9bc-1f80-11e7-9dcd-b20a3e339e33.png)
![image](https://cloud.githubusercontent.com/assets/4991075/24972335/ba6fc366-1f80-11e7-9e1a-1f6c8a58129e.png)

![image](https://cloud.githubusercontent.com/assets/4991075/24981321/db20ef38-1fa1-11e7-95f9-4f70a643686b.png)
![image](https://cloud.githubusercontent.com/assets/4991075/24981332/e61cb62e-1fa1-11e7-964c-c9c4e63f9ebe.png)

**Comment by larryscheier** [comment]
In SimAirServingZones.cc, the Minimum Zone Ventilation Efficiency field is applied in LimitZoneVentEff. It looks like the VRP table "Zone Ventilation Calculations for Cooling Design" is not getting updated for the columns: "Minimum Zone Primary Airflow - Vpz-min [m3/s]" and "Primary Outdoor Air Fraction - Zpz" though somehow the "Zone Ventilation Efficiency - Evz" column is updated. Perhaps we should update these columns in this table as noted below:

```
LimitZoneVentEff(
if ( SystemCoolingEv < FinalZoneSizing( CtrlZoneNum ).ZoneVentilationEff ) {
	// reset ZoneOAFrac
	ZoneOAFrac = 1.0 + Xs - FinalZoneSizing( CtrlZoneNum ).ZoneVentilationEff;
	// reset AvailSAFlow (which in this case is minimum cooling supply air flow rate)
	AvailSAFlow = Voz/ZoneOAFrac;
	// save ZoneOAFrac 
	FinalZoneSizing( CtrlZoneNum ).ZpzClgByZone = ZoneOAFrac;
	// save new (increased) minimum flow rate
	FinalZoneSizing( CtrlZoneNum ).DesCoolVolFlowMin = AvailSAFlow;
	// set the system ventilation efficiency to the user specified minimum
	SystemCoolingEv = FinalZoneSizing( CtrlZoneNum ).ZoneVentilationEff;_

**Suggested new code:
	// Update VRP table entries:
	PreDefTableEntry( pdchS62zcdVpzmin, FinalZoneSizing( CtrlZoneNum ).ZoneName, VpzMinClgByZone( CtrlZoneNum ), 4 ); //Vpz-min
	PreDefTableEntry( pdchS62zcdZpz, FinalZoneSizing( CtrlZoneNum ).ZoneName, FinalZoneSizing( CtrlZoneNum ).ZpzClgByZone, 3 ); //Zpz = Voz/Vpz


```

**Comment by mjwitte** [comment]
What I learned today.
1.  It's not clear that the suggestion to add PreDefTableEntry calls in LimitZoneVentEff will work to change the table values (for one thing VpcMinClgByZone array is already gone by this point).  We may still want the table to change, but it won't change anything in the VAV terminal unit min flow rate.
2.  Part of the problem lies [here in SingleDuct::SizeSys](https://github.com/NREL/EnergyPlus/blob/11c37e3b2f2a2dd9d70ed2d9638e27b26069835f/src/EnergyPlus/SingleDuct.cc#L2001-L2031).
If Sizing:Zone "Cooling Minimum Air Flow Fraction" is non-blank, then it is used to set the VAV min flow fraction (or rate).  If it *is* blank, then the other default calculations, including VRP adjustments to `FinalZoneSizing( CurZoneEqNum ).DesCoolVolFlowMin` will get used.
3.  *Rather than checking if Sizing:Zone "Cooling Minimum Air Flow Fraction" is blank, it seems that SizeSys should just use the greater of that and DesCoolVolFlowMin.*

**Comment by mjwitte** [comment]
What I did today.

  1. Yes, the suggested PreDefTableEntry calls in LImitZoneVentEff are good.  And two more were also added.
  2. The VAV min flow sizing dependency on whether SizingZone "Cooling Minimum Air Flow Fraction" is blank has been removed.
  3. These changes along with some doc updates are now in #6140.

**Comment by Myoldmopar** [comment]
Fixed via #6140
